Source/CesiumRuntime/Private/CesiumBoundingVolumeComponent.cpp

```diff
@@ -0,0 +1,160 @@
+// Copyright 2020-2021 CesiumGS, Inc. and Contributors
+
+#include "CesiumBoundingVolumeComponent.h"
+#include "CalcBounds.h"
+#include "CesiumGeoreference.h"
+#include "CesiumLifetime.h"
+#include "UObject/UObjectGlobals.h"
+#include "VecMath.h"
+#include <optional>
+#include <variant>
+
+using namespace Cesium3DTilesSelection;
+
+UCesiumBoundingVolumePoolComponent::UCesiumBoundingVolumePoolComponent()
+    : _cesiumToUnreal(1.0) {
+  this->_pPool = std::make_shared<UCesiumBoundingVolumePool>(this);
+  // SetMobility(EComponentMobility::Static);
+  SetMobility(EComponentMobility::Movable);
+}
+
+TileOcclusionRendererProxy* UCesiumBoundingVolumePoolComponent::createProxy() {
+  UCesiumBoundingVolumeComponent* pBoundingVolume =
+      NewObject<UCesiumBoundingVolumeComponent>(this);
+  pBoundingVolume->SetVisibility(false);
+  pBoundingVolume->bUseAsOccluder = false;
+  pBoundingVolume->SetUsingAbsoluteLocation(true);
+
+  // TODO: check if we can get away with this, may need to explicitly recreate
+  // the scene proxy when georeference changes, reassigned to a different tile,
+  // etc.
+  // pBoundingVolume->SetMobility(EComponentMobility::Static);
+  pBoundingVolume->SetMobility(EComponentMobility::Movable);
+  pBoundingVolume->SetFlags(
+      RF_Transient | RF_DuplicateTransient | RF_TextExportTransient);
+  pBoundingVolume->SetupAttachment(this);
+  pBoundingVolume->RegisterComponent();
+
+  pBoundingVolume->UpdateTransformFromCesium(this->_cesiumToUnreal);
+
+  return (TileOcclusionRendererProxy*)pBoundingVolume;
+}
+
+void UCesiumBoundingVolumePoolComponent::destroyProxy(
+    TileOcclusionRendererProxy* pProxy) {
+  // TODO: verify correctness of destruction
```

Comment:
I think this will be fine so long as the ConditionalBeginDestroy detaches it from its parent. It's not obvious from a quick glance at the code whether this is the case, though.